astrosat_users/views/views_customers.py

```diff
@@ -63,6 +59,24 @@ def get_queryset(self):
         return Customer.objects.multiple()
 
 
+class CustomerListView(CustomerViewMixin, generics.CreateAPIView):
+
+    lookup_field = "id"
+    lookup_url_kwarg = "customer_id"
+
+    permission_classes = [IsAuthenticated]
```

Comment:
Should this not be limited to `Admins`/`Managers`? Why would we want anyone else to get a list of customers or customer users? Not arguing, just not sure why we wouldn't want to limit it further.

Reply:
So, I think that my naming convention has confused you.  And that indicates to me that it's probably wrong.  In general, in Django-Rest-Framework, you use `DetailView` when you're referring to a single object the URL and `ListView` when you aren't referring to a single object in the URL.  So...

* `CustomerListView`: "/api/customers/"
* `CustomerDetailView`: "/api/customers/<customer_id>/"

However, I'm not actually exposing "GET" on the `ListView` only "POST"; I only allow users to _create_ a `customer` not to `list` them.  The other view exposes "GET" and "PUT/PATCH" but not "DELETE"; I only allow users to _retrieve_ and _update_ customers not to _delete_ them.

So perhaps these should be renamed to `CustomerCreateView` and `CustomerUpdateView`?

Reply:
_(And `CustomerListView` shouldn't be limited to `admins`/`managers` b/c anybody can create a `customer`)_

Reply:
Okay, so now you've explained this, I see how you are limiting the views using `generics` overrides, so `generics.CreateAPIView` just allows `POST` and `generics.RetrieveUpdateAPIView`, just gives `GET` and `PUT/PATCH`. This makes sense. I still think the renaming is a good idea, but if it goes against a convention you have enacted elsewhere, I'm concerned renaming this will create just as much confusion. Maybe I just have to learn your convention is all.